extensions-core/avro-extensions/src/main/java/org/apache/druid/data/input/avro/InlineSchemaAvroBytesDecoder.java

```diff
@@ -88,7 +88,6 @@ public GenericRecord parse(ByteBuffer bytes)
       return reader.read(null, DecoderFactory.get().binaryDecoder(inputStream, null));
     }
     catch (EOFException eof) {
-      // waiting for avro v1.9.0 (#AVRO-813)
```

Comment:
If this is fixed, let's remove the error handling code in addition to removing the comment. And let's also double-check that the issue was in fact fixed. (Hopefully, we have a unit test that verifies this without extra work on our part right now.)

Reply:
Unfortunately, the error still appears in 1.11.1 sporadically - that's how I stumbled into this stale comment trying to find the source of exception. Not sure what the cause is though. Reading through the [issue](https://issues.apache.org/jira/browse/AVRO-813) and other related forums, it seems like `EOFException` can still occur in legitimate scenarios where a producer didn't close a file descriptor correctly or when there's a connection reset. Perhaps the `ParseException` could add some more details?

Reply:
Hmm. Do you have more info on the `EOFException` you saw?

One of those cases you mentioned (connection reset), I don't see how it can happen. I don't think our decoders make network calls inside this `try` block. At least I hope they don't! If they do we should try to adjust it so they don't.

You mentioned another case too (producer didn't close properly). Maybe in that case we get corrupt data, in which case, yeah, a `ParseException` is appropriate.

If that's what's happening, not a network call, then IMO it makes sense to remove the `EOFException` block and have an `IOException` block that wraps in a more normal-looking `ParseException` that doesn't reference AVRO-813.

Reply:
Had the chance to look a bit more closely - here's a stacktrace:
```
java.io.EOFException: null
	at org.apache.avro.io.BinaryDecoder.readBoolean(BinaryDecoder.java:164) ~[?:?]
	at org.apache.avro.io.ValidatingDecoder.readBoolean(ValidatingDecoder.java:76) ~[?:?]
	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:204) ~[?:?]
	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161) ~[?:?]
	at org.apache.avro.generic.GenericDatumReader.readField(GenericDatumReader.java:260) ~[?:?]
	at org.apache.avro.generic.GenericDatumReader.readRecord(GenericDatumReader.java:248) ~[?:?]
	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:180) ~[?:?]
	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161) ~[?:?]
	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:154) ~[?:?]
	at org.apache.druid.data.input.avro.InlineSchemaAvroBytesDecoder.parse(InlineSchemaAvroBytesDecoder.java:88) ~[?:?]
	at org.apache.druid.data.input.avro.AvroStreamReader.intermediateRowIterator(AvroStreamReader.java:78) ~[?:?]
```

Tracing it to the avro [code](https://github.com/apache/avro/blob/release-1.11.1/lang/java/avro/src/main/java/org/apache/avro/io/BinaryDecoder.java#L164). Also, the `read()` method in `1.11.1` now [only](https://avro.apache.org/docs/1.11.1/api/java/org/apache/avro/util/ByteBufferInputStream.html) throws `IOException` (in [previous](https://avro.apache.org/docs/1.7.6/api/java/org/apache/avro/util/ByteBufferInputStream.html) versions, it seems like both `EOFException` and `IOException` were thrown). So yeah, I will  replace the `EOFException` catch block with `IOException` and update the exception message.

Reply:
ok, done, thanks for the discussion 🙂